File: app/src/androidTest/java/com/sample/lmn/davide/getoutofwork/TimeSchedulPresenterEspressoTest.java
package com.sample.lmn.davide.getoutofwork;

import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import com.sample.lmn.davide.getoutofwork.managers.RealmPersistenceManager;
import com.sample.lmn.davide.getoutofwork.presenters.TimeScheduleRegisterPresenter;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import io.realm.Realm;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withText;
import static java.util.Calendar.AM;
import static junit.framework.Assert.assertEquals;
import static org.hamcrest.Matchers.allOf;

/**
 * Created by davide-syn on 7/5/17.
 */

@RunWith(AndroidJUnit4.class)
public class TimeSchedulPresenterEspressoTest {
    @Rule
    public ActivityTestRule<MainActivity> activityTestRule = new ActivityTestRule<MainActivity>(MainActivity.class);
    private TimeScheduleRegisterPresenter presenter;

    @Before
    public void init() {
        //check in to make in error setCheckIn
        Realm.init(activityTestRule.getActivity());
        presenter = new TimeScheduleRegisterPresenter(activityTestRule.getActivity(),
                new RealmPersistenceManager(Realm.getDefaultInstance()));
    }

    @Test
    public void testCheckInAm() {
        assertEquals(presenter.setCheckInAm(), false);
        //find snackbar with that text
        onView(allOf(withId(android.support.design.R.id.snackbar_text), withText(activityTestRule.getActivity().getString(R.string.generic_error) + AM)))
                .check(matches(isDisplayed()));
    }
}

File: app/src/androidTest/java/com/sample/lmn/davide/getoutofwork/TimeSchedulPresenterEspressoTest.kt
package com.sample.lmn.davide.getoutofwork

import android.support.test.InstrumentationRegistry
import android.support.test.espresso.ViewInteraction
import android.support.test.runner.AndroidJUnit4
import com.sample.lmn.davide.getoutofwork.managers.RealmPersistenceManager
import com.sample.lmn.davide.getoutofwork.models.TimeScheduleRealm
import com.sample.lmn.davide.getoutofwork.presenters.TimeScheduleRegisterPresenter
import com.sample.lmn.davide.getoutofwork.ui.views.TimeScheduleRegisterView
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import java.util.*

/**
 * Created by davide-syn on 7/5/17.
 */

@RunWith(AndroidJUnit4::class)
class TimeSchedulPresenterEspressoTest {
    lateinit var presenter: TimeScheduleRegisterPresenter

    @Before
    fun init() {
        //check in to make in error setCheckIn
        InstrumentationRegistry.getInstrumentation().runOnMainSync {
            presenter = TimeScheduleRegisterPresenter(object: TimeScheduleRegisterView {
                override fun setClockOutTime(date: Date?) {
                }

                override fun showErrorUI(message: String) {
                }

                override fun updateCheckCardview(timeSchedule: TimeScheduleRealm) {
                }
            }, RealmPersistenceManager.Holder.instance)
        }
    }

    @Test
    fun testCheckEmptySnackbar() {
        //find snackbar with that text
//        onViewErrorSnackbar(activityTestRule.activity.getString(R.string.generic_error) + AM).check(matches(isDisplayed()))
    }

    @Test
    fun testCheckIn() {
            presenter.setCheck()
        //find snackbar with that text
//        onViewErrorSnackbar(activityTestRule.activity.getString(R.string.generic_error) + AM).check(matches(isDisplayed()))

    }
    @Test
    fun testCheckOut() {
//            presenter.setCheckAndDateTime(OutInEnum.OUT, AM)
            presenter.setCheck()
        //find snackbar with that text
//        onViewErrorSnackbar(activityTestRule.activity.getString(R.string.generic_error) + PM).check(matches(isDisplayed()))
    }

    @Test
    fun testValidSetCheckDateAM_OUT() {

    }

    @Test
    fun testValidSetCheckDateAM_IN() {

    }
    @Test
    fun testValidSetCheckDatePM_OUT() {
    }

    @Test
    fun testValidSetCheckDatePM_IN() {
    }

    private fun onViewErrorSnackbar(text: String): Unit {
//        return onView(allOf<View>(withId(androidx.R.id.snackbar_text), withText(text)))
    }

}
